olp-cpp-sdk-core/tests/client/OlpClientTest.cpp

```diff
@@ -368,6 +368,122 @@ TEST_P(OlpClientTest, RetryTimeExponential) {
   }
 }
 
+TEST_P(OlpClientTest, RetryWithExponentialBackdownStrategy) {
+  const std::chrono::milliseconds::rep kInitialBackdownPeriod = 100;  // msec
+  size_t expected_retry_count = 0;
+  std::vector<std::chrono::milliseconds::rep> wait_times = {
+      kInitialBackdownPeriod};
+
+  // Setup retry settings:
+  client_settings_.retry_settings.initial_backdown_period =
+      kInitialBackdownPeriod;
+
+  client_settings_.retry_settings.retry_condition =
+      ([](const olp::client::HttpResponse&) { return true; });
+
+  client_settings_.retry_settings.backdown_strategy =
+      [kInitialBackdownPeriod, &expected_retry_count, &wait_times](
+          std::chrono::milliseconds period,
+          size_t retry_count) -> std::chrono::milliseconds {
+    EXPECT_EQ(kInitialBackdownPeriod, period.count());
+    EXPECT_EQ(++expected_retry_count, retry_count);
+
+    const auto wait_time = olp::client::ExponentialBackdownStrategy()(
+        std::chrono::milliseconds(kInitialBackdownPeriod), retry_count);
+    wait_times.push_back(wait_time.count());
+    return wait_time;
+  };
+
+  // previous version of backdown policy shouldn't be called,
+  // if the new backdown policy was specified
+  client_settings_.retry_settings.backdown_policy = [](int period) -> int {
+    // we can't invoke FAIL directly from the callback:
```

Comment:
It's possible to add a non-fatal failure using ADD_FAILURE_AT which doesn't require a wrapping lambda.